hyperspy/io_plugins/emd.py

```diff
@@ -1100,8 +1100,13 @@ def _read_image(self, image_group, image_sub_group_key):
             # Change back to standard API once issue #977 is fixed.
             # Preallocate the numpy array and use read_direct method, which is
             # much faster in case of chunked data.
-            data = np.empty(h5data.shape)
+            data = np.empty(h5data.shape, h5data.dtype)
```

Comment:
> but the 1.5 MB is a bit large

Will cut into this one and try to erase the larger file from commit history. I made the data 3x3x25 to keep FrameLookupTable the same as in the experimental file, but will cut the number of "frames".

> fft was calculated from were real? Is so, I think this is fine to load as it is and document it accordingly.

Ok, just with a comment in the code or add a warning to the user also? The data will not look like it does in Velox basically when plotted, which might confuse some users. Then again, I suppose most users will make their own fft's from the image data and not rely on the one from Velox.

> I would simply load it non-lazily even lazy=True

Fair point, I will implement it like that. Lazy reading works, but when I attempt plot or compute it fails because of the odd data type. I will add a logger warning that lazy doesn't work for FFT.

> It seems that defining the dtype here make the read_direct call much slower for some dataset

Wow you are right, the difference is really order of magnitude. Very odd. I will rework it so that it only happens if the dtype is special.

Reply:
> > fft was calculated from were real? Is so, I think this is fine to load as it is and document it accordingly.
> 
> Ok, just with a comment in the code or add a warning to the user also? The data will not look like it does in Velox basically when plotted, which might confuse some users. Then again, I suppose most users will make their own fft's from the image data and not rely on the one from Velox.

Maybe add a few sentences in the user guide (https://hyperspy.readthedocs.io/en/latest/user_guide/io.html#emd-velox) to explain that the FFT is not centered and only the positive frequency are saved in the file?
In any case, as you said, users would be better off calculating the FFT again.

Reply:
done